mina-payouts-data-provider/scripts/ledger-db-bootstrap.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+
+# Check if environment variables are set
+check_env_variable() {
+    if [ -z "${!1}" ]; then
+        echo "The $1 environment variable is not set or is empty."
+        exit 1
+    fi
+}
+
+# Check required environment variables
+check_env_variable "LEDGER_DB_QUERY_USER"
+check_env_variable "LEDGER_DB_QUERY_PASSWORD"
+check_env_variable "LEDGER_DB_QUERY_HOST"
+check_env_variable "LEDGER_DB_QUERY_PORT"
+check_env_variable "LEDGER_DB_QUERY_NAME"
+check_env_variable "LEDGER_DB_QUERY_SCHEMA_URL"
+check_env_variable "LEDGER_DB_QUERY_BOOTSTRAP_DELAY"
+check_env_variable "LEDGER_DB_QUERY_REQUIRE_SSL"
+
+PG_CONN=postgres://${LEDGER_DB_QUERY_USER}:${LEDGER_DB_QUERY_PASSWORD}@${LEDGER_DB_QUERY_HOST}:${LEDGER_DB_QUERY_PORT}/${LEDGER_DB_QUERY_NAME}
+
+# Check SSL requirements
+if [ "${LEDGER_DB_QUERY_REQUIRE_SSL}" = "true" ]; then
+    check_env_variable "LEDGER_DB_QUERY_CERTIFICATE"
+    # Dump certificate to a file
+    CERTIFICATE_FILE="/tmp/certificate.crt"
```

Comment:
Here I would advise to introduce some randomness in the name of the variable, maybe PID of the process with $$.

`/tmp` is writeable by all users, and if a malicious person wanted us to execute their `.crt` or any of the temp files that are executed here, he could inject them in the right place and the script would pick it up.

Reply:
Or use `mktemp`

Reply:
You are right that it is difficult to execute attacks with ephemeral jobs, and that attackers might have bigger fish to fry, I just wanted to underline the possibility of an attack angle.

If you don't think it imposes a security risk that is fine by me.